src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.models.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HibernateUtil;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        return userDao.add(user);
+    }
+
+    @Override
+    public User findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
+            Root<User> userRoot = criteriaQuery.from(User.class);
+            Predicate emailPredicate = criteriaBuilder.equal(userRoot.get("email"), email);
+            criteriaQuery.select(userRoot).where(emailPredicate);
+            return session.createQuery(criteriaQuery).getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t find user by login", e);
+        }
```

Comment:
What is the **_DAO_** layer for and what is the **_SERVICE_** layer for?

Also, there is no need to use the catch block since we return the **_Optional_**
If we don't find the required user by such an email we return **_Optional.empty()_**
Then if we try to extract the value and there is no such found we receive a **_NullPointerException_**